benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -975,25 +1133,240 @@ void GraphicsBenchmarkApp::DrawExtraInfo()
+        grfx::ShaderModuleCreateInfo shaderCreateInfo = {static_cast<uint32_t>(bytecode.size()), bytecode.data()};
+        PPX_CHECKED_CALL(GetDevice()->CreateShaderModule(&shaderCreateInfo, &blit.vs));
+
+        bytecode = LoadShader("basic/shaders", "FullScreenTriangle.ps");
+        PPX_ASSERT_MSG(!bytecode.empty(), "PS shader bytecode load failed");
+        shaderCreateInfo = {static_cast<uint32_t>(bytecode.size()), bytecode.data()};
+        PPX_CHECKED_CALL(GetDevice()->CreateShaderModule(&shaderCreateInfo, &blit.ps));
+
+        grfx::FullscreenQuadCreateInfo createInfo = {};
+        createInfo.VS                             = blit.vs;
+        createInfo.PS                             = blit.ps;
+        createInfo.setCount                       = 1;
+        createInfo.sets[0].set                    = 0;
+        createInfo.sets[0].pLayout                = blit.descriptorSetLayout;
+        createInfo.renderTargetCount              = 1;
+        createInfo.renderTargetFormats[0]         = GetSwapchain()->GetColorFormat();
+        createInfo.depthStencilFormat             = GetSwapchain()->GetDepthFormat();
+
+        PPX_CHECKED_CALL(GetDevice()->CreateFullscreenQuad(&createInfo, &blit.quad));
+    }
+
+    return ppx::SUCCESS;
+}
+
+GraphicsBenchmarkApp::RenderPasses GraphicsBenchmarkApp::OffscreenRenderPasses(const OffscreenFrame& frame)
+{
+    return RenderPasses{
+        frame.loadRenderPass,
+        frame.clearRenderPass,
+        frame.noloadRenderPass,
+        nullptr,
+        nullptr,
+        nullptr,
+        &frame,
+    };
+}
+
+void GraphicsBenchmarkApp::DestroyOffscreenFrame(OffscreenFrame& frame)
+{
+    GetDevice()->DestroyRenderPass(frame.loadRenderPass);
+    GetDevice()->DestroyRenderPass(frame.clearRenderPass);
+    GetDevice()->DestroyRenderPass(frame.noloadRenderPass);
+
+    for (auto& rtv : frame.renderTargetViews) {
+        GetDevice()->DestroyRenderTargetView(rtv);
+    }
+    GetDevice()->DestroyDepthStencilView(frame.depthStencilView);
+
+    GetDevice()->FreeDescriptorSet(frame.blitDescriptorSet);
+    GetDevice()->DestroyTexture(frame.blitSource);
+
+    GetDevice()->DestroyImage(frame.colorImage);
+    GetDevice()->DestroyImage(frame.depthImage);
+
+    // Reset all the pointers to nullptr.
+    frame = {};
+}
+
+ppx::Result GraphicsBenchmarkApp::CreateOffscreenFrame(OffscreenFrame& frame, grfx::Format colorFormat, grfx::Format depthFormat, uint32_t width, uint32_t height)
+{
+    frame = OffscreenFrame{width, height, colorFormat, depthFormat};
+    {
+        grfx::ImageCreateInfo colorCreateInfo   = grfx::ImageCreateInfo::RenderTarget2D(width, height, colorFormat);
+        colorCreateInfo.initialState            = grfx::RESOURCE_STATE_PRESENT;
+        colorCreateInfo.usageFlags.bits.sampled = true;
+        ppx::Result ppxres                      = GetDevice()->CreateImage(&colorCreateInfo, &frame.colorImage);
+        if (ppxres != ppx::SUCCESS) {
```

Comment:
Could these checks be `PPX_CHECKED_CALL` instead?

Reply:
Those error should be handled, we might encounter format/size that some platform does not support.
I've not add the error handling logic since I have not found a device that does not support the format/size yet (mostly testing on Nvidia&AMD's desktop gpus).